components/component.go

```diff
@@ -80,12 +83,18 @@ type ManifestsConfig struct {
 	SourcePath string `json:"sourcePath,omitempty"`
 }
 
+type ComponentReleaseDetails struct {
+	ComponentVersion semver.Version
+	RepositoryURL    string
+}
+
 type ComponentInterface interface {
 	Init(ctx context.Context, platform cluster.Platform) error
-	ReconcileComponent(ctx context.Context, cli client.Client,
-		owner metav1.Object, DSCISpec *dsciv1.DSCInitializationSpec, platform cluster.Platform, currentComponentStatus bool) error
+	ReconcileComponent(ctx context.Context, cli client.Client, owner client.Object, DSCISpec *dsciv1.DSCInitializationSpec,
```

Comment:
any reason to change to client.Object for this PR?

Reply:
it was probably a consequence of some discussions about moving the logic in the `ReconcileComponent` but I think it should now be reverted.